src/test/CalldataLogic.t.sol

```diff
@@ -0,0 +1,284 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.10;
+
+import {DataTypes} from './../../contracts/protocol/libraries/types/DataTypes.sol';
+import {ReserveConfiguration} from './../../contracts/protocol/libraries/configuration/ReserveConfiguration.sol';
+
+import {CalldataLogic} from './../../contracts/protocol/libraries/logic/CalldataLogic.sol';
+import {L2Encoder} from './../../contracts/misc/L2Encoder.sol';
+
+import {Vm} from './../Vm.sol';
+import {TestHelper} from './TestHelper.sol';
+
+// TODO: There is an error with the revert messages. For now, we are checking after reverts only,
+// as the integration tests are already catching the revert messages
+contract CalldataLogicTest is TestHelper {
+  using ReserveConfiguration for DataTypes.ReserveConfigurationMap;
+
+  Vm constant VM = Vm(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
```

Comment:
DSTest has [`HEVM_ADDRESS`](https://github.com/dapphub/ds-test/blob/0a5da56b0d65960e6a994d2ec8245e6edd38c248/src/test.sol#L41) as a constant so you can use that directly for readability. You can also declared this `VM` variable just once in `TestHelper` instead of in each individual test contract 🙂

```suggestion
  Vm constant VM = Vm(HEVM_ADDRESS);
```